iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/thread.inl

```diff
@@ -0,0 +1,40 @@
+#ifndef IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+#define IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+
+#include "iceoryx_hoofs/posix_wrapper/thread.hpp"
+
+namespace iox
+{
+namespace posix
+{
+inline cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitializedThread,
+                                                        const cxx::function<void()>& callable) noexcept
+{
+    if (!callable)
+    {
+        std::cerr << "The thread cannot be created with an empty callable." << std::endl;
+        return cxx::error<ThreadError>(ThreadError::EMPTY_CALLABLE);
+    }
+
+    uninitializedThread.emplace();
```

Comment:
This shows again some shortcomings with the API to pass an `optional<thread>` to the builder. It is a source of a potential nullable type in an optional which has a value. This could happen on an early return when another feature is added and the developer forgets to reset the `optional`.